src/rulesets/Base/indexing.jl

```diff
@@ -1,3 +1,48 @@
+#####
+##### getindex(::Tuple)
+#####
+
+function frule((_, ẋ), ::typeof(getindex), x::Tuple, i::Integer)
+    return x[i], ẋ[i]
+end
+
+function frule((_, ẋ), ::typeof(getindex), x::Tuple, i)
+    @show i x[i]
+    return x[i], Tangent{typeof(x)}(ẋ[i]...)
+end
+
+function rrule(::typeof(getindex), x::Tuple, i::Integer)
+    proj = ProjectTo(x)
+    len = Val(length(x))
+    getindex_back_1(dy) = (NoTangent(), proj(ntuple(j -> j == i ? dy : NoTangent(), len)), NoTangent())
+    return x[i], getindex_back_1
+end
+
+function rrule(::typeof(getindex), x::NTuple{N,T}, i::Integer) where {N, T<:Number}
+    proj = ProjectTo(x)
+    len = Val(N)
+    function getindex_back_2(dy_raw)
+        dy = unthunk(dy_raw)
+        dx = ntuple(j -> j == i ? dy : zero(dy), len)
```

Comment:
Not certain it's worthwhile, but the idea of this was to allow uniform tuples to have type-stable behaviour.

Zygote does not do this, always uses `nothing` for others. Has a special method for ranges of indices though.

Reply:
my instict is that this is not worth doing